### PR TITLE
[LR-01] Implement cross validation and least square

### PR DESCRIPTION
**Pull request by spandan12**

### CODE REVIEW

File: readfile.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import pandas as pd
+import numpy as np
+
+
+def read_file(file_path):
+
+    return pd.read_csv(file_path, sep=" ", header=None).dropna(axis="columns")
```
Comment (bipin-lekhak) [inline readfile.py]: The separator seems to be 3 spaces or something of that like in the inputs, the dropna handles this specific case but that might not be the best way to do it. Why not just us the multispaced separator?

File: leastsquare.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+
+def calculate_weight(basis_function, output_labels):
+
+    # For MW = S
+    S = np.dot(basis_function, output_labels)
```
Comment (bipin-lekhak) [inline leastsquare.py]: Another way to do this is as below, not sure which is more pythonic:
```python
S = basis_function@output_labels
```

File: leastsquare.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    M = np.dot(basis_function, np.transpose(basis_function))
```
Comment (bipin-lekhak) [inline leastsquare.py]: Another way to do this is as below, not sure which is more pythonic:
```python
M = basis_function@basis_function.T
```

File: leastsquare.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    # Solve linear equation
+    return np.linalg.solve(M, S)
+
+
+def calculate_error(basis_function, output_labels, weight):
+    predicted_output = np.dot(np.transpose(weight), basis_function)
```
Comment (bipin-lekhak) [inline leastsquare.py]: Another way to do this is as below, not sure which is more pythonic:
```python
predicted_output = weigth.T@basis_function
```

File: leastsquare.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    rms = np.sum((predicted_output - np.transpose(output_labels)) ** 2)
```
Comment (bipin-lekhak) [inline leastsquare.py]:
```python
rms = np.norm( predicted_output - output_labels.T ) ** 2
```

File: crossvalidate.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import math
```
Comment (bipin-lekhak) [inline crossvalidate.py:1]: unused import

**Review by bipin-lekhak — CHANGES_REQUESTED** [review]
Some changes recommended.

File: readfile.py (hunk 1 above)
Comment (spandan12) [reply readfile.py]: That's a good idea.

File: leastsquare.py (hunk 2 above)
Comment (spandan12) [reply leastsquare.py]: Simplified implementation. Implemented

File: leastsquare.py (hunk 3 above)
Comment (spandan12) [reply leastsquare.py]: Simplified implementation. Implemented

File: leastsquare.py (hunk 4 above)
Comment (spandan12) [reply leastsquare.py]: Simplified implementation. Implemented

File: crossvalidate.py (hunk 6 above)
Comment (spandan12) [reply crossvalidate.py:1]: resolved

**Pull request merged**